app/main/forms.py

```diff
@@ -274,7 +287,12 @@ class GovukEmailField(GovukTextInputFieldMixin, EmailField):
 
 class GovukSearchField(GovukTextInputFieldMixin, SearchField):
     input_type = "search"
-    param_extensions = {"classes": "govuk-!-width-full"}  # email addresses don't need to be spellchecked
+    param_extensions = {"classes": "govuk-!-width-full"}
+
+
+class GovukInlineButtonSearchField(GovukInlineButtonTextInputFieldMixin, SearchField):
+    input_type = "search"
+    param_extensions = {"button": {"classes": "govuk-!-width-full notify-input--inline-button", "text": "Search"}}
```

Comment:
this `text` is never used as we define custom html to pass in to param_extensions when we actually use this class in search.html
```suggestion
    param_extensions = {"button": {"classes": "govuk-!-width-full notify-input--inline-button"}}
```

Reply:
I think I'm OK keeping this nonetheless, as this field could be re-used and without it the button will just have no text at all, which doesn't feel like a sensible default.